File: src/main/java/mnm/mods/tabbychat/client/gui/component/IGuiEventListenerDelegate.java
package mnm.mods.tabbychat.client.gui.component;

import net.minecraft.client.gui.IGuiEventListener;

public interface IGuiEventListenerDelegate extends IGuiEventListener {

    IGuiEventListener delegate();

    @Override
    default void mouseMoved(double p_212927_1_, double p_212927_3_) {
        delegate().mouseMoved(p_212927_1_, p_212927_3_);
    }

    @Override
    default boolean mouseClicked(double p_mouseClicked_1_, double p_mouseClicked_3_, int p_mouseClicked_5_) {
        return delegate().mouseClicked(p_mouseClicked_1_, p_mouseClicked_3_, p_mouseClicked_5_);
    }

    @Override
    default boolean mouseReleased(double p_mouseReleased_1_, double p_mouseReleased_3_, int p_mouseReleased_5_) {
        return delegate().mouseReleased(p_mouseReleased_1_, p_mouseReleased_3_, p_mouseReleased_5_);
    }

    @Override
    default boolean mouseDragged(double p_mouseDragged_1_, double p_mouseDragged_3_, int p_mouseDragged_5_, double p_mouseDragged_6_, double p_mouseDragged_8_) {
        return delegate().mouseDragged(p_mouseDragged_1_, p_mouseDragged_3_, p_mouseDragged_5_, p_mouseDragged_6_, p_mouseDragged_8_);
    }

    @Override
    default boolean mouseScrolled(double p_mouseScrolled_1_, double p_mouseScrolled_3_, double p_mouseScrolled_5_) {
        return delegate().mouseScrolled(p_mouseScrolled_1_, p_mouseScrolled_3_, p_mouseScrolled_5_);
    }

    @Override
    default boolean keyPressed(int p_keyPressed_1_, int p_keyPressed_2_, int p_keyPressed_3_) {
        return delegate().keyPressed(p_keyPressed_1_, p_keyPressed_2_, p_keyPressed_3_);
    }

    @Override
    default boolean keyReleased(int p_keyReleased_1_, int p_keyReleased_2_, int p_keyReleased_3_) {
        return delegate().keyReleased(p_keyReleased_1_, p_keyReleased_2_, p_keyReleased_3_);
    }

    @Override
    default boolean charTyped(char p_charTyped_1_, int p_charTyped_2_) {
        return delegate().charTyped(p_charTyped_1_, p_charTyped_2_);
    }

    @Override
    default boolean changeFocus(boolean p_changeFocus_1_) {
        return delegate().changeFocus(p_changeFocus_1_);
    }

    @Override
    default boolean isMouseOver(double p_isMouseOver_1_, double p_isMouseOver_3_) {
        return delegate().isMouseOver(p_isMouseOver_1_, p_isMouseOver_3_);
    }
}

File: src/main/kotlin/mnm/mods/tabbychat/client/gui/component/IGuiEventListenerDelegate.kt
package mnm.mods.tabbychat.client.gui.component

import net.minecraft.client.gui.IGuiEventListener

@Deprecated("Use a kotlin delegate type")
interface IGuiEventListenerDelegate : IGuiEventListener  {

    @Deprecated("Use a kotlin delegate type")
    fun delegate(): IGuiEventListener

    override fun mouseMoved(p_212927_1_: Double, p_212927_3_: Double) {
        delegate().mouseMoved(p_212927_1_, p_212927_3_)
    }

    override fun mouseClicked(p_mouseClicked_1_: Double, p_mouseClicked_3_: Double, p_mouseClicked_5_: Int): Boolean {
        return delegate().mouseClicked(p_mouseClicked_1_, p_mouseClicked_3_, p_mouseClicked_5_)
    }

    override fun mouseReleased(p_mouseReleased_1_: Double, p_mouseReleased_3_: Double, p_mouseReleased_5_: Int): Boolean {
        return delegate().mouseReleased(p_mouseReleased_1_, p_mouseReleased_3_, p_mouseReleased_5_)
    }

    override fun mouseDragged(p_mouseDragged_1_: Double, p_mouseDragged_3_: Double, p_mouseDragged_5_: Int, p_mouseDragged_6_: Double, p_mouseDragged_8_: Double): Boolean {
        return delegate().mouseDragged(p_mouseDragged_1_, p_mouseDragged_3_, p_mouseDragged_5_, p_mouseDragged_6_, p_mouseDragged_8_)
    }

    override fun mouseScrolled(p_mouseScrolled_1_: Double, p_mouseScrolled_3_: Double, p_mouseScrolled_5_: Double): Boolean {
        return delegate().mouseScrolled(p_mouseScrolled_1_, p_mouseScrolled_3_, p_mouseScrolled_5_)
    }

    override fun keyPressed(p_keyPressed_1_: Int, p_keyPressed_2_: Int, p_keyPressed_3_: Int): Boolean {
        return delegate().keyPressed(p_keyPressed_1_, p_keyPressed_2_, p_keyPressed_3_)
    }

    override fun keyReleased(p_keyReleased_1_: Int, p_keyReleased_2_: Int, p_keyReleased_3_: Int): Boolean {
        return delegate().keyReleased(p_keyReleased_1_, p_keyReleased_2_, p_keyReleased_3_)
    }

    override fun charTyped(p_charTyped_1_: Char, p_charTyped_2_: Int): Boolean {
        return delegate().charTyped(p_charTyped_1_, p_charTyped_2_)
    }

    override fun changeFocus(p_changeFocus_1_: Boolean): Boolean {
        return delegate().changeFocus(p_changeFocus_1_)
    }

    override fun isMouseOver(p_isMouseOver_1_: Double, p_isMouseOver_3_: Double): Boolean {
        return delegate().isMouseOver(p_isMouseOver_1_, p_isMouseOver_3_)
    }
}
